watchtower/wtpolicy/policy.go

```diff
@@ -71,3 +73,30 @@ func (p Policy) String() string {
 		"sweep-fee-rate=%d)", p.BlobType, p.MaxUpdates, p.RewardRate,
 		p.SweepFeeRate)
 }
+
+// ComputeAltruistOutput computes the lone output value of a justice transaction
+// that pays no reward to the tower. The value is computed using the weight of
+// of the justice transaction and subtracting an amount that satisfies the
+// policy's fee rate.
+func (p *Policy) ComputeAltruistOutput(totalAmt btcutil.Amount,
+	txWeight int64) (btcutil.Amount, error) {
+
+	txFee := p.SweepFeeRate.FeeForWeight(txWeight)
+	if txFee > totalAmt {
+		return 0, ErrFeeExceedsInputs
+	}
+
+	sweepAmt := totalAmt - txFee
+
+	// TODO(conner): replace w/ configurable dust limit
+	dustLimit := lnwallet.DefaultDustLimit()
+
+	// Check that the created outputs won't be dusty.
+	if sweepAmt <= dustLimit {
+		return 0, ErrCreatesDust
+	}
+
+	// TODO(conner): also compare output value to fee?
```

Comment:
Compare to check that we're not throwing away most of it to the fee?

Reply:
yep i left the todo here, but i'm thinking it should be moved to a higher level as this threshold wouldn't be a policy shared by both client and server. it would be purely client-level configuration